### DISCUSSION TITLE
关于单元测试 @umi/test

### DISCUSSION THREAD

**Discussion by wangzhongl** [umijs/umi#11413] [Q&A]
关于单元测试 @umi/test 官方是不是已经不在支持内置了

**Comment by fz6m** [comment]
由于 esbuild 的一些更新，不再能容易的 hack jest 来测试 ts 文件了，这个已知问题还一直没解。

目前比较推荐方法是用 vitest ，可以单测逻辑，或者做纯净组件的测试（和 umi 逻辑无关的），[这个例子](https://github.com/umijs/umi/tree/master/examples/test-vitest) 可以参考。

如果涉及到 umi 逻辑的组件需要测试，可以考虑用 `<TestBrowser />` 包裹来提供 umi 上下文环境，参考 [这个例子](https://github.com/umijs/umi/blob/c31259b13f71f7c649b2b9796d7253ef360e4d44/examples/ant-design-pro/src/pages/user/Login/index.test.tsx#L16) ，但这个上下文还是不够完善，如果要测试组件的话，更推荐测试纯净的组件，可以通过 props 传递相关参数。

**Comment by wangzhongl** [comment]
请问为啥我运行了官方配置jest.config.ts配置下的jest命令后 在src目录下自动生成了一个.umi.test文件 但是在这之后运行项目时 会报找不到该模块 
![image](https://github.com/umijs/umi/assets/103638520/dc82cf5d-a023-43a2-934f-148dc50adcca)

error - container entry:./.umi-test[0]
Module not found: Error: Can't resolve 'D:/ewan/eworld-library/src/.umi-test' in 'D:\ewan\eworld-library'

**Reply by fz6m** [reply]
如上文所述，现在更推荐使用 vitest 做测试。

**Comment by wangzhongl** [comment]
jest.config.ts配置如下：
![image](https://github.com/umijs/umi/assets/103638520/54221be2-9778-4a86-8203-63a774853931)

**Comment by wangzhongl** [comment]
我的项目目录最外层是dumi 我想请问下内置的配置 可移动到dumi那一层去嘛
![image](https://github.com/umijs/umi/assets/103638520/a060aa3b-89ac-46ce-be7c-085d80280e7c)
